flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/sources/CsvTableSource.java

```diff
@@ -101,6 +101,40 @@ public CsvTableSource(
 			quoteCharacter, ignoreFirstLine, ignoreComments, lenient);
 	}
 
+	/**
+	 * A {@link InputFormatTableSource} and {@link LookupableTableSource} for simple CSV files with
+	 * a (logically) unlimited number of fields.
+	 *
+	 * @param path            	The path to the CSV file.
+	 * @param fieldNames      	The names of the table fields.
+	 * @param fieldTypes      	The types of the table fields.
+	 * @param fieldDelim      	The field delimiter, "," by default.
+	 * @param lineDelim       	The row delimiter, "\n" by default.
+	 * @param quoteCharacter  	An optional quote character for String values, null by default.
+	 * @param ignoreFirstLine 	Flag to ignore the first line, false by default.
+	 * @param ignoreComments  	An optional prefix to indicate comments, null by default.
+	 * @param lenient         	Flag to skip records with parse error instead to fail, false by
+	 *                        	default.
+	 * @param emptyColumnAsNull	Flag to treat empty column as null.
+	 */
+	public CsvTableSource(
```

Comment:
How about we deprecating all existing constructors and to encourage users using `Builder` instead? We can have only one unified private constructor then.

Reply:
Sounds good, importing the new two constructors just for  compatibility.